### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by Iangyl**
Closes: https://github.com/polywrap/polyfolio/issues/94

### CODE REVIEW

File: src/common/components/AssetTransaction/useTransactions.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import useGetData from 'common/hooks/useGetData/useGetData';
+
+const useTransactions = () => {
+  const state = useRecoilValue(transactionState);
+  const formateData = useGetData();
+  const preparedData = formateData();
+  const {user} = useAuth();
+  const data = [];
+
+  console.log(state);
```
Comment (DmitriyIvchenko) [inline src/common/components/AssetTransaction/useTransactions.ts:24]: remove

File: src/common/components/AssetsTable/AssetsTable.tsx [hunk 2]
```diff
@@ -51,7 +49,7 @@ function AssetsTable() {
         filter={filter.assets}
         onChange={onChange}
         isOpen={isOpen}
-        sum={sum}
+        sum={preparedData['allAssetsSum']}
```
Comment (DmitriyIvchenko) [inline src/common/components/AssetsTable/AssetsTable.tsx:52]: 1. what is preparedData? maybe assetTableData?

2. why not preparedData. allAssetsSum

File: src/common/components/AssetsTable/AssetsTable.tsx [hunk 3]
```diff
@@ -26,16 +28,12 @@ function AssetsTable() {
   const theme = useTheme();
   const translation = useTranslation();
   const assets = useAssets();
+  const formatedData = useGetData(page);
+  const preparedData = formatedData();
```
Comment (DmitriyIvchenko) [inline src/common/components/AssetsTable/AssetsTable.tsx:32]: same here

File: src/common/components/AssetsTable/AssetsTableItem/AssetsTableItem.types.ts [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
   iconInfoPage?: string;
-  valueTitle?: number;
-  priceTitle?: number;
-  percent?: number;
+  valueTitle: string;
+  priceTitle: string;
+  percent: string;
   title: string;
   icon?: string;
   link?: string;
```
Comment (DmitriyIvchenko) [inline src/common/components/AssetsTable/AssetsTableItem/AssetsTableItem.types.ts:14]: why items are optional?

File: src/common/components/HeaderDashboard/HeaderDashboard.tsx [hunk 5]
```diff
@@ -28,7 +31,7 @@ function HeaderDashboard() {
       <div className={styles.contentContainer}>
         <div>
           <span className={styles.secondaryTitle}>{translation.Dashboard.secondaryTitle}</span>
-          <h2>${numberFormatter({value: content.title, size: 2})}</h2>
+          <h2>${numberFormatter({value: preparedData['allAssetsSum'], size: 2})}</h2>
```
Comment (DmitriyIvchenko) [inline src/common/components/HeaderDashboard/HeaderDashboard.tsx:34]: same

File: src/common/components/HeaderDashboard/HeaderDashboard.tsx [hunk 6]
```diff
@@ -10,12 +10,15 @@ import Dropdown from '../Dropdown/Dropdown';
...
 import useTranslation from 'common/hooks/useTranslation/useTranslation';
+import useGetData from 'common/hooks/useGetData/useGetData';
 
 function HeaderDashboard() {
   const [isOpen, setIsOpen] = useState<boolean>(false);
   const [currency, setCurrency] = useState(dropdownItems[0]);
   const theme = useTheme();
   const translation = useTranslation();
+  const formatedData = useGetData();
+  const preparedData = formatedData();
```
Comment (DmitriyIvchenko) [inline src/common/components/HeaderDashboard/HeaderDashboard.tsx:21]: same

File: src/common/components/HeaderDashboard/HeaderDashboard.types.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 export interface DashboardTypes {
-  title?: number;
+  title?: string;
   id?: number;
-  value?: number;
-  percent?: number;
+  value?: string;
+  percent?: string;
```
Comment (DmitriyIvchenko) [inline src/common/components/HeaderDashboard/HeaderDashboard.types.ts:5]: same

File: src/common/components/HeaderGasInfoMenu/HeaderGasInfoMenu.types.ts [hunk 8]
```diff
@@ -1,9 +1,9 @@
 export interface HeaderGasInfoItem {
   title?: string;
   time?: string;
-  price?: number;
+  price?: string;
   colorIcon?: string;
-  titlePrice?: number;
+  titlePrice?: string;
   isDivider?: boolean;
   isExternal?: boolean;
```
Comment (DmitriyIvchenko) [inline src/common/components/HeaderGasInfoMenu/HeaderGasInfoMenu.types.ts:8]: same

File: src/common/components/Networks/Networks.config.ts [hunk 9]
```diff
@@ -1,21 +1,30 @@
...
 
 const useNetworks = () => {
-  const balance = useRecoilValue(balanceState);
+  const formateData = useGetData();
+  const preparedData = formateData();
   const menuItems: NetworksItem[] = [];
-
-  if (balance) {
+  
+  if (preparedData['balance']) {
```
Comment (DmitriyIvchenko) [inline src/common/components/Networks/Networks.config.ts:15]: what is this preparedData['balance']? why not preparedData.balance?

File: src/common/components/Networks/Networks.config.ts [hunk 10]
```diff
@@ -1,21 +1,30 @@
...
+import { rmCommasFromNum } from 'utils/helpers';
 import _ from 'lodash';
-import { useRecoilValue } from 'recoil';
-import {networks} from 'utils/constants';
+import { ejectAssetsFromProtocol, getAssetsValueSum } from 'utils/dataFormating';
 
 const useNetworks = () => {
-  const balance = useRecoilValue(balanceState);
+  const formateData = useGetData();
+  const preparedData = formateData();
```
Comment (DmitriyIvchenko) [inline src/common/components/Networks/Networks.config.ts:12]: same

File: src/common/components/Networks/Networks.config.ts [hunk 11]
```diff
@@ -1,21 +1,30 @@
...
   const menuItems: NetworksItem[] = [];
-
-  if (balance) {
+  
+  if (preparedData['balance']) {
     networks.forEach(item => {
+      let allAssets = [];
+      _.forEach(preparedData['balance'][item.name]?.protocols, protocol => {
+        allAssets = _.flatten([...allAssets, ...ejectAssetsFromProtocol(protocol)]);
```
Comment (DmitriyIvchenko) [inline src/common/components/Networks/Networks.config.ts:19]: different methods?

File: src/common/components/Networks/Networks.config.ts [hunk 12]
```diff
@@ -1,21 +1,30 @@
...
+  const formateData = useGetData();
+  const preparedData = formateData();
   const menuItems: NetworksItem[] = [];
-
-  if (balance) {
+  
+  if (preparedData['balance']) {
     networks.forEach(item => {
+      let allAssets = [];
+      _.forEach(preparedData['balance'][item.name]?.protocols, protocol => {
```
Comment (DmitriyIvchenko) [inline src/common/components/Networks/Networks.config.ts:18]: what is preparedData['balance'][item.name]? where is types?

File: src/common/components/Networks/Networks.config.ts [hunk 13]
```diff
@@ -1,21 +1,30 @@
...
-  const balance = useRecoilValue(balanceState);
+  const formateData = useGetData();
+  const preparedData = formateData();
   const menuItems: NetworksItem[] = [];
-
-  if (balance) {
+  
+  if (preparedData['balance']) {
     networks.forEach(item => {
+      let allAssets = [];
```
Comment (DmitriyIvchenko) [inline src/common/components/Networks/Networks.config.ts:17]: I can't understand why do we need this?

File: src/common/components/PerformanceTable/PerformanceItem.types.ts [hunk 14]
```diff
@@ -1,5 +1,5 @@
 export interface PerformanceItem {
-  title?: number;
-  secondaryTitleValue?: number;
-  secondaryTitlePercent?: number;
+  title?: string;
+  secondaryTitleValue?: string;
+  secondaryTitlePercent?: string;
```
Comment (DmitriyIvchenko) [inline src/common/components/PerformanceTable/PerformanceItem.types.ts:4]: same

File: src/common/components/Profile/Profile.tsx [hunk 15]
```diff
@@ -25,6 +26,8 @@ function Profile() {
   const translation = useTranslation();
   const {user, logOut} = useAuth();
   const {connect} = useWallet();
+  const formatedData = useGetData();
+  const preparedData = formatedData();
```
Comment (DmitriyIvchenko) [inline src/common/components/Profile/Profile.tsx:30]: same

File: src/common/components/VaultsTable/VaultsTable.tsx [hunk 16]
```diff
@@ -23,6 +28,8 @@ function ProtocolsTable({totalTableValue}: {totalTableValue: number}) {
   const {filters, setFilters} = useFiltersTables();
   const [filter, setFilter] = useState<Filters>(filters);
   const menuItems = GetVaults();
+  const formateData = useGetData(page);
+  const preparedData = formateData();
```
Comment (DmitriyIvchenko) [inline src/common/components/VaultsTable/VaultsTable.tsx:32]: same

File: src/common/components/VaultsTable/VaultsTable.tsx [hunk 17]
```diff
@@ -42,7 +49,7 @@ function ProtocolsTable({totalTableValue}: {totalTableValue: number}) {
         menuFields={menuFields}
         onChange={onChange}
         isOpen={isOpen}
-        sum={totalTableValue}
+        sum={preparedData['allAssetsSum']}
```
Comment (DmitriyIvchenko) [inline src/common/components/VaultsTable/VaultsTable.tsx:52]: same

File: src/common/components/VaultsTable/VaultsTableItem/VaultsTableItem.config.ts [hunk 18]
```diff
@@ -1,27 +1,27 @@
...
+import useGetData from 'common/hooks/useGetData/useGetData';
+import { useLocation } from 'react-router-dom';
 
 export const GetVaults = () => {
-  const protocol = useRecoilValue(currentProtocol);
-  const assetsTotalValue = getAssetsValueSum(protocol?.assets);
+  const { pathname } = useLocation()
+  const page = getStringFromPath(pathname, 1);
+  const formateData = useGetData(page);
+  const preparedData = formateData();
```
Comment (DmitriyIvchenko) [inline src/common/components/VaultsTable/VaultsTableItem/VaultsTableItem.config.ts:11]: same

File: src/common/components/VaultsTable/VaultsTableItem/VaultsTableItem.config.ts [hunk 19]
```diff
@@ -1,27 +1,27 @@
...
 
-  return _.map(protocol?.assets, asset => {
+  return _.map(preparedData['allAssets'], asset => {
```
Comment (DmitriyIvchenko) [inline src/common/components/VaultsTable/VaultsTableItem/VaultsTableItem.config.ts:13]: same

File: src/common/hooks/useBalance/useBalance.ts [hunk 20]
```diff
@@ -3,20 +3,18 @@ import {useRecoilState} from 'recoil';
...
   const client = useWeb3ApiClient();
 
   const [balance, setBalance] = useRecoilState(balanceState);
-  const [allAssetsSum, setAllAssetsSum] = useRecoilState(allAssetsSumState);
 
-  const balanceRequest = useCallback(async (name) => {
+  const balanceRequest = useCallback(async (chainId) => {
     const { data: response, errors } = await client.query({
-      uri: `ipfs/QmRYP5qwQd7AotVbtcx7KhN8HuHX9DCg8sS9LVE4kstpVw`,
+      uri: `ens/rinkeby/mock.defiwrapper.eth`,
```
Comment (DmitriyIvchenko) [inline src/common/hooks/useBalance/useBalance.ts:17]: move to config

File: src/common/hooks/useGetData/useGetData.ts [hunk 21]
```diff
@@ -0,0 +1,36 @@
+import {useRecoilValue} from 'recoil';
+import balanceState from 'common/modules/atoms/balanceState';
+import networkDataFormating from 'utils/networkDataFormating';
+import allNetworksDataFormating from 'utils/allNetworksDataFormating';
+
+const useGetData = (name?: string) => {
+  const balance = useRecoilValue(balanceState);
+
+  const formateData = () => {
```
Comment (DmitriyIvchenko) [inline src/common/hooks/useGetData/useGetData.ts:9]: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

File: src/common/hooks/useTransaction/useTransaction.ts [hunk 22]
```diff
@@ -14,40 +13,55 @@ export default function useTransactions() {
...
       )
-    }`,
+    }
+    `,
     config: {
       envs: [
         {
-          uri: 'ens/rinkeby/mock.defiwrapper.eth',
-          common: {
+          uri: "ens/rinkeby/ethereum.token.resolvers.defiwrapper.eth",
```
Comment (DmitriyIvchenko) [inline src/common/hooks/useTransaction/useTransaction.ts:27]: move to config

File: src/common/hooks/useTransaction/useTransaction.ts [hunk 23]
```diff
@@ -14,40 +13,55 @@ export default function useTransactions() {
...
     },
   });
 
   const getTransactions = useCallback(async () => {
     if (user && !loading && !data) {
       const {data: response, errors} = await execute({
-        accountAddress: user,
-        vsCurrency: 'USDT',
+        account: user,
+        currency: 'USD',
```
Comment (DmitriyIvchenko) [inline src/common/hooks/useTransaction/useTransaction.ts:58]: is this correct? why not currency from picker?

File: src/common/hooks/useTransaction/useTransaction.ts [hunk 24]
```diff
@@ -14,40 +13,55 @@ export default function useTransactions() {
...
       });
   
       if (response && !errors?.length) {
         const transactions = response?.getTransactions;
-        console.log(transactions);
 
+        console.log(transactions)
```
Comment (DmitriyIvchenko) [inline src/common/hooks/useTransaction/useTransaction.ts:64]: remove

File: src/utils/dataFormating.ts [hunk 25]
```diff
@@ -1,17 +1,120 @@
...
+      } 
+    });
+  });
+}
 
 export const getAssetsValueSum = (assets) => {
   if (assets) {
     return _.sumBy(assets, (value) => 
-      _.round(Number(rmCommasFromNum(value['balance'].token.values[0].value)), 2)
+      _.round(Number(rmCommasFromNum(value['token'].values[0].value)), 2)
```
Comment (DmitriyIvchenko) [inline src/utils/dataFormating.ts:21]: value.token

**Review by DmitriyIvchenko — CHANGES_REQUESTED** [review]

File: src/common/hooks/useTransaction/useTransaction.ts (hunk 22 above)
Comment (Iangyl) [reply src/common/hooks/useTransaction/useTransaction.ts:27]: what config do you mean?

File: src/utils/dataFormating.ts (hunk 25 above)
Comment (Iangyl) [reply src/utils/dataFormating.ts:21]: does not exist on property unknown

**Review by DmitriyIvchenko — APPROVED** [review]

**Pull request merged**